### PR TITLE
Salt electrolysis

### PR DESCRIPTION
**Pull request by Sakamoto4ka**
## About The Pull Request

Add salt electrolysis. You need to mix ethereal blood with table salt/sodium chlorine/NaCl and you get sodium and chlorine. 

## Why It's Good For The Game

Add ghetto chemistry way to obtain sodium

Add new reaction for ghetto chimestry

## Changelog

:cl:
add: Added salt electrolysis reaction
/:cl:

### CODE REVIEW

**Comment by Sadboysuss** [comment]
I like it, but how does it work ingame? Does the ethereal blood work like a stabilising agent to prevent the Na and Cl from mixing into table salt again?

**Comment by zoomachina** [comment]
It's immediately going to turn into salt again

**Comment by Sakamoto4ka** [comment]
Fixed. Now ethereal blood is catalyst and prevent salt form reaction.

**Comment by CCC23** [comment]
thats what I said sodium chloride

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 1]
```diff
@@ -38,6 +38,12 @@
 	results = list(/datum/reagent/consumable/salt = 2)
 	required_reagents = list(/datum/reagent/sodium = 1, /datum/reagent/chlorine = 1) // That's what I said! Sodium Chloride!
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_FOOD
+	required_other = TRUE
+
+/datum/chemical_reaction/sodiumchloride/pre_reaction_other_checks(datum/reagents/holder)
+	if(holder.has_reagent(/datum/reagent/consumable/liquidelectricity) || holder.has_reagent(/datum/reagent/consumable/liquidelectricity/enriched))
+		return FALSE
+	. = ..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/others.dm:46]: is this right?

```suggestion
/datum/chemical_reaction/sodiumchloride/pre_reaction_other_checks(datum/reagents/holder)
	. = ..()	
	if(holder.has_reagent(/datum/reagent/consumable/liquidelectricity) || holder.has_reagent(/datum/reagent/consumable/liquidelectricity/enriched))
		return FALSE

```

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -600,6 +606,20 @@
...
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_CHEMICAL
+
+//salt electrolysis
+/datum/chemical_reaction/saltelectrolysis
+	results = list(/datum/reagent/chlorine = 2.5, /datum/reagent/sodium = 2.5)
+	required_reagents = list(/datum/reagent/consumable/salt = 5)
+	required_catalysts = list(/datum/reagent/consumable/liquidelectricity = 1)
+	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_CHEMICAL
+
+/datum/chemical_reaction/saltelectrolysis2
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/others.dm:617]: subtype this and just change required catalysts?

either that or use required_other, dont like 2 reactions for the same thing if we needed to change it down the line.

**Comment by Sakamoto4ka** [comment]
Where is ma points

**Comment by Sadboysuss** [comment]
> Where is ma points

feature is -5gbp

**Comment by Sakamoto4ka** [comment]
i need ma first good pr points

**Pull request merged**